### PR TITLE
Initial refactor of TS tests

### PR DESCRIPTION
**Pull request by willsoto**
Opening for visibility and feedback. Already noticing some gaps in coverage so I'll be trying to fix those as I work through these.

Ref #1431

### CODE REVIEW

File: tests/ts/query-examples/basic-queries/delete.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class Person extends objection.Model {}
+
+(async () => {
+  await Person.query().deleteById(1);
+
+  const numDeleted: number = await Person.query()
+    .delete()
+    .where(raw('lower("firstName")'), 'like', '%ennif%');
+
+  console.log(numDeleted, 'people were deleted');
```
Comment (mordof) [inline tests/ts/query-examples/basic-queries/delete.ts]: i don't think these will ever actually be run, will they?

Comment (mordof) [reply tests/ts/query-examples/basic-queries/delete.ts]: hm i guess it doesn't matter if these files are purely for giving some examples

**Comment by mordof** [comment]
There's already an `examples` folder. can we move these new examples into `examples/ts`?

**Comment by koskimas** [comment]
@willsoto The idea here seems to be to copy the examples from the documentation? They are in no way comprehensive, but I guess that's a good place to start. I'd rather have real unit tests like `where.ts` that tests all the possible ways you can call `where/orWhere/andWhere/etc.`.

File: tests/ts/query-examples/basic-queries/delete.ts (hunk 1 above)
Comment (willsoto) [reply tests/ts/query-examples/basic-queries/delete.ts]: Yeah. For right now I'm just blindly copy-pasting from the example. I don't mind either way on these but they don't add any value, so might just remove.

Comment (willsoto) [reply tests/ts/query-examples/basic-queries/delete.ts]: Yeah. For right now I'm just blindly copy-pasting from the example. I don't mind either way on these but they don't add any value, so might just remove.

**Comment by willsoto** [comment]
> There's already an examples folder. can we move these new examples into examples/ts?

I see purpose of the tests as different from what is happening in the `examples` folders. The `examples` are more "end-to-end" what does it look like to work with Objection whereas the tests are just going through the many possible forms of queries, models, etc.

With that said, I took a glance at the `examples` folder and it looks like it could maybe be updated.

> The idea here seems to be to copy the examples from the documentation? They are in no way comprehensive, but I guess that's a good place to start. I'd rather have real unit tests like `where.ts` that tests all the possible ways you can call `where/orWhere/andWhere/etc.`.

That is right. And it is just a starting point. The kind of coverage you are describing is exactly where I want to go with this. I will also try and move what makes sense from the existing "tests" to the appropriate locations since we know those are battle-tested.

**Comment by willsoto** [comment]
I haven't forgotten about this. Been busy with other things but I'll have some time to get back to this soon.

**Comment by koskimas** [comment]
I merged new-typings with v2.0. You can point this to v2.0 branch now.

**Comment by willsoto** [comment]
I've rebased and added tests for:

- [Basic Queries](https://vincit.github.io/objection.js/guide/query-examples.html#basic-queries)
- [Relation Queries](https://vincit.github.io/objection.js/guide/query-examples.html#relation-queries)

I am not sure how you prefer to proceed. It might be good to merge these PRs in little by little to prevent something that is 1000+ lines long. Happy to do either, but ultimately your call.

If you feel like this is good enough to merge, please go ahead and do so and I'll make follow-up PRs.

PS: I am not sure why tests are failing only on Node 12. I can take a look at fixing unless you already have a fix in mind?

**Comment by willsoto** [comment]
@koskimas pinging to get your thoughts on my previous comment

**Comment by koskimas** [comment]
We can merge this as soon as the tests pass

**Comment by willsoto** [comment]
@koskimas looks like tests are passing now 👍

**Pull request merged**